+nosnoc/+discrete_time_problem/Heaviside.m

```diff
@@ -399,36 +456,113 @@ function generate_direct_transcription_constraints(obj)
+                                relax_p = vdx.RelaxationStruct(opts.relax_terminal_physical_time.to_vdx, 's_physical_time', 'rho_physical_time');
+                                obj.g.sum_h = {sum_h_all-obj.p.T(), relax_n};
+                                obj.g.integral_clock = {sum_h_all-obj.w.T_final(), relax_p};
+                                if relax_n.is_relaxed
+                                    obj.p.rho_numerical_time().val = opts.rho_terminal_numerical_time;
+                                end
+                                if relax_p.is_relaxed
+                                    obj.p.rho_physical_time().val = opts.rho_terminal_physical_time;
+                                end
+                            end
+                        end
+                    end
+                end
+            end
             
             % Terminal constraint
             if opts.relax_terminal_constraint_homotopy
                 error("Currently unsupported")
             end
             g_terminal = dcs.g_terminal_fun(x_end, z_end, v_global, p_global);
-            switch opts.relax_terminal_constraint
-              case ConstraintRelaxationMode.NONE % hard constraint
-                if opts.relax_terminal_constraint_from_above
-                    obj.g.terminal = {g_terminal, model.lbg_terminal, inf*ones(dims.n_g_terminal,1)};
-                else
-                    obj.g.terminal = {g_terminal, model.lbg_terminal, model.ubg_terminal};
-                end
-              case ConstraintRelaxationMode.ELL_1 % l_1
-                obj.w.s_terminal_ell_1 = {{'s_terminal_ell_1', dims.n_g_terminal}, 0, inf, 10};
-
-                g_terminal = [g_terminal-model.lbg_terminal-obj.w.s_terminal_ell_1();
-                    -(g_terminal-model.ubg_terminal)-obj.w.s_terminal_ell_1()];
-                obj.g.terminal = {g_terminal, -inf, 0};
-                obj.f = obj.f + obj.p.rho_terminal_p()*sum(obj.w.s_terminal_ell_1());
-              case ConstraintRelaxationMode.ELL_2 % l_2
-                                                  % TODO(@anton): this is as it was implemented before. should handle lb != ub?
-                obj.f = obj.f + obj.p.rho_terminal_p()*(g_terminal-model.lbg_terminal)'*(g_terminal-model.lbg_terminal);
-              case ConstraintRelaxationMode.ELL_INF % l_inf
-                obj.w.s_terminal_ell_inf = {{'s_terminal_ell_inf', 1}, 0, inf, 1e3};
-
-                g_terminal = [g_terminal-model.lbg_terminal-obj.w.s_terminal_ell_inf();
-                    -(g_terminal-model.ubg_terminal)-obj.w.s_terminal_ell_inf()];
-                obj.g.terminal = {g_terminal, -inf, 0};
-                obj.f = obj.f + obj.p.rho_terminal_p()*obj.w.s_terminal_ell_inf();
+            relax = vdx.RelaxationStruct(opts.relax_terminal_constraint.to_vdx, 's_terminal', 'rho_terminal');
```

Comment:
nice simplification, but as said above, a one-line comment should explain the constraint relaxation syntax

Reply:
possibly add a link to the vdx docu or example where it is introduced (if it exists)

Reply:
This is an excellent idea, however I have not updated `vdx` docs recently. Will put it in vdx issues.